ballerina/websocket_service.bal

```diff
@@ -25,10 +25,10 @@ isolated service class WsService {
     private final readonly & __Schema schema;
     private final Context context;
 
-    isolated function init(Engine engine, __Schema & readonly schema) {
+    isolated function init(Engine engine, __Schema & readonly schema, Context context) {
         self.engine = engine;
         self.schema = schema;
-        self.context = new;
+        self.context = context;
```

Comment:
Why do we need to store the context in the service? It does not make any sense.

Reply:
AFAIK, When a user initializes a WebSocket connection, only the first request goes through the Upgrade Service. Then we can access the `http: Request` and create the context object. After the initialization of the connection, other requests go through the WebSocket remote functions. Since context can not be created after the initial request, store the context here to use for the rest of the messages in that connection.

Reply:
That is wrong. The WebSocket service is initialized once for the listener. How are you going to handle multiple requests with this approach? 
A context is a request-specific object. You cannot create it per service, or cannot use a single context object for multiple requests. That's the whole point of the issue you are trying to fix here.